src/components/buttons/icon-button/icon-button.spec.js

```diff
@@ -45,4 +46,34 @@ describe('rendering', () => {
     const { getByTestId } = render(<IconButton {...props} />);
     expect(getByTestId('icon')).toBeInTheDocument();
   });
+  describe('when used with `as`', () => {
+    describe('when as is a valid HTML element', () => {
+      it('should render as that HTML element', () => {
+        const { container } = render(
+          <IconButton
+            {...props}
+            as="a"
+            href="https://www.kanyetothe.com"
+            target="_BLANK"
```

Comment:
if we don't support passing all valid props, then using regular "external" links would be super annoying.